tests/ut_devicelock/ut_devicelock.cpp

```diff
@@ -17,289 +17,475 @@
+/** Go through all states that affect device lock
+ */
+void Ut_DeviceLock::testAllTransitions(void)
+{
+    /* Go through all possible combinations of state variables */
+    for (size_t lockingDelayIndex = 0; lockingDelayIndex < numof(lockingDelays); ++lockingDelayIndex) {
+        applyLockingDelay(lockingDelayIndex);
+
+        for (size_t callStateIndex = 0; callStateIndex < numof(callStates); ++callStateIndex) {
+            applyCallState(callStateIndex);
+
+            for (size_t displayStateIndex = 0; displayStateIndex < numof(displayStates); ++displayStateIndex) {
+                applyDisplayState(displayStateIndex);
+
+                for (size_t tklockStateIndex = 0; tklockStateIndex < numof(tklockStates); ++tklockStateIndex) {
+                    applyTklockState(tklockStateIndex);
+
+                    for (size_t inactivityStateIndex = 0; inactivityStateIndex < numof(inactivityStates); ++inactivityStateIndex) {
+                        applyInactivityState(inactivityStateIndex);
+
+                        for (size_t deviceLockStateIndex = 0; deviceLockStateIndex < numof(deviceLockStates); ++deviceLockStateIndex) {
+                            applyDeviceLockState(deviceLockStateIndex);
+
+                            /* Simulate all one step transitions that can be made
+                             * from each state. */
+                            iterateDeviceLockStates(deviceLockStateIndex);
+                            iterateLockingDelays(lockingDelayIndex);
+                            iterateCallStates(callStateIndex);
+                            iterateDisplayStates(displayStateIndex);
+                            iterateTklockStates(tklockStateIndex);
+                            iterateInactivityStates(inactivityStateIndex);
+                        }
+                    }
+                }
+            }
+        }
+    }
 }
 
-void Ut_DeviceLock::testStateOnAutomaticLockingAndTouchScreenLockState()
+void Ut_DeviceLock::testSignalConnections()
 {
-    QFETCH(int, lockingDelayValue);
-    QFETCH(MeeGo::QmLocks::State, touchScreenLockState);
-    QFETCH(DeviceLock::LockState, deviceLockState);
+    QCOMPARE(disconnect(deviceLock->m_hbTimer, SIGNAL(running()),
+                        deviceLock, SLOT(lock())), true);
+    // FIXME: Should the dbus signal listener connections be tested?
+}
 
-    deviceLock->setState(DeviceLock::Locked);
+void Ut_DeviceLock::cleanup()
+{
+    delete deviceLock; deviceLock = 0;
+}
 
-    deviceLock->lockingDelay = lockingDelayValue;
-    gQmLocksStub->stubSetReturnValue("getState", touchScreenLockState);
-    deviceLock->setStateAndSetupLockTimer();
+void Ut_DeviceLock::init()
+{
+    deviceLock = new DeviceLock();
 
-    QCOMPARE(deviceLock->state(), (int)deviceLockState);
+    /* In real use we want to have some transition logging, but
+     * for unit tests they just generate huge amounts of noise */
+    deviceLock->m_verbosityLevel = false;
```

Comment:
Hmm, looks like this could be done with QLoggingCategory, too. Setup a lipstick.devicelock logging category in devicelock.cpp, with "debug" as default logging level:
```
Q_LOGGING_CATEGORY(LIPSTICK_LOG_DEVICELOCK, "lipstick.devicelock", QtDebugMsg)
```

Then, here raise the default logging level:
```
QLoggingCategory::setFilterRules("lipstick.devicelock.debug=false");
```

And for debug output use 
```
qCDebug(LIPSTICK_LOG_DEVICELOG) << "yessir";
```

For future reference, that is.